ConfigSpace/hyperparameters.pyx

```diff
@@ -85,6 +85,47 @@ cdef class Hyperparameter(object):
         vector = self._sample(rs)
         return self._transform(vector)
 
+    def rvs(self, size: Optional[int]=1, random_state: Optional[Union[int, np.random, np.random.RandomState]]=None) -> Union[float, np.ndarray]:
+        """
+        scipy compatibility wrapper for ``_sample``,
+        allowing the hyperparameter to be used in sklearn API
+        hyperparameter searchers, eg. GridSearchCV.
+
+        """
+
+        # copy-pasted from scikit-learn utils/validation.py
+        def check_random_state(seed):
+            """
+            Turn seed into a np.random.RandomState instance
+            If seed is None (or np.random), return the RandomState singleton used
+            by np.random.
+            If seed is an int, return a new RandomState instance seeded with seed.
+            If seed is already a RandomState instance, return it.
+            If seed is a new-style np.random.Generator, return it.
+            Otherwise, raise ValueError.
+
+            """
+            if seed is None or seed is np.random:
+                return np.random.mtrand._rand
+            if isinstance(seed, (int, np.integer)):
+                return np.random.RandomState(seed)
+            if isinstance(seed, np.random.RandomState):
+                return seed
+            try:
+                # Generator is only available in numpy >= 1.17
+                if isinstance(seed, np.random.Generator):
+                    return seed
+            except AttributeError:
+                pass
+            raise ValueError('%r cannot be used to seed a numpy.random.RandomState'
+                            ' instance' % seed)
+
+        vector = self._sample(rs=check_random_state(random_state), size=size)
```

Comment:
Hey, what do you think about using `self.sample` instead of `self._sample` followed by `self.transform`? That looks more compact.

Reply:
`self.sample()` doesn't support `size`, unlike `self._sample()`. I thought that'd be cleaner than using a loop or something similar.

Reply:
That makes sense, thanks for explaining.